test/framework/provider/wait.go

```diff
@@ -52,27 +65,41 @@ func WaitForAllProvidersInstalled(ctx context.Context, c client.Client, interval
 }
 
 // Wait for Provider to be successfully updated.
-func WaitForRevisionTransition(ctx context.Context, c client.Client, p2 string, p1 string, interval time.Duration, timeout time.Duration) error {
+func WaitForRevisionTransition(ctx context.Context, c client.Client, p2 string, p1 string, interval time.Duration, timeout time.Duration, t *testing.T) error {
 	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
 		l := &v1.ProviderRevisionList{}
 		if err := c.List(ctx, l); err != nil {
 			return false, err
 		}
+		// t.Logf("Provider Revisions list: %+v", l.Items)
+		for _, item := range l.Items {
+			t.Logf("Name: %v \t Type: %v", item.Name, item.TypeMeta)
+		}
+
 		// There should be a revision present for the initial revision and the upgrade.
 		if len(l.Items) != 2 {
+			t.Log("The no. of providers revisions is not equal to 2")
+			for i, item := range l.Items{
+				t.Logf("Provider revision %v : %v uses package %v", i, item.Name, item.Spec.Package)
+			}
 			return false, nil
 		}
-		for _, p := range l.Items {
+		for _, item := range l.Items {
 			// New ProviderRevision should be Active.
-			if p.Spec.Package == p2 && p.GetDesiredState() != v1.PackageRevisionActive {
+			if item.Spec.Package == p2 && item.GetDesiredState() != v1.PackageRevisionActive {
+				t.Logf("The state of new provider revision %v having version %v is %v", item.Name, item.ObjectMeta.ResourceVersion, item.Status)
+				//TODO
+				// add on which p are we looking at using t.Logf.. for all statements
```

Comment:
Has this TODO been addressed?
```suggestion
```

Reply:
Yes, this todo is addressed as the value we receive from `item.Name` gives us which provider revision we are working with.  Just forgot to remove the additional comments that I had kept in the file for my own reference. Removed them.